eml/screens/lectures/VideoLectureScreen.jsx

```diff
@@ -0,0 +1,188 @@
+import React, { useState, useEffect, useRef } from 'react';
+
+// For animating play button
+
+import { View, Pressable } from 'react-native';
+
+import Text from '../../components/general/Text';
+
+import VideoActions from '../../components/lectures/VideoActions';
+import { MaterialCommunityIcons } from '@expo/vector-icons';
+
+
+import CustomExpoVideoPlayer from '../../components/lectures/VideoPlayer';
+import ReactSliderProgress from './ReactSliderProgress';
+import { getVideoDownloadUrl } from '../../api/api';
+
+import { useNavigation } from '@react-navigation/native';
+
+
+export default function VideoLectureScreen({ lecture, course, progress }) {
+
+
+    const videoRef = useRef(null);
+    const [isPlaying, setIsPlaying] = useState(false); // Keep track of playback status
+    const [positionMillis, setPositionMillis] = useState(0);
+    const [durationMillis, setDurationMillis] = useState(0);
+    const [isMuted, setIsMuted] = useState(false); // Keep track of mute status
+
+
+    const onStatusUpdate = (status) => {
+        setPositionMillis(status.positionMillis || 0);
+        setDurationMillis(status.durationMillis || 0);
+
+    };
+
+
+
+    const [videoUrl, setVideoUrl] = useState(null);
+
+    useEffect(() => {
+        const _videoUrl = getVideoDownloadUrl(lecture.video, "180p")
+
+
+
+        //test if video is available for download from internet
+
+        setVideoUrl(_videoUrl)
+    }, [])
+
+
+    useEffect(() => {
+        if (videoRef.current) {
+        }
+    }, [videoRef]);
+
+    const handlePress = () => {
+
+
+        if (!videoRef.current) {
+            return;
+        }
+
+
+        setIsPlaying(!isPlaying);
+    };
+
+    const handleMutepress = () => {
+
+        setIsMuted(!isMuted);
+    }
+
+    const navigation = useNavigation();
+
+    //check if video url is valid
+    useEffect(() => {
+        const _videoUrl = getVideoDownloadUrl(lecture._id, "180p");
+
+        //check for video url validity maybe not needed
+        // fetch(_videoUrl, {
+        //     method: 'HEAD'
+        // })
+        //     .then(response => {
+        //         if (response.ok) {
+        //             // HTTP status between 200-299 or equals 304.
+        //             setVideoUrl(_videoUrl);
+        //         } else {
+        //             console.error('Video URL is not valid');
+        //             Alert.alert("Error", "The video is corrupted. Please try again later", "OK");
+        //         }
+        //     })
+        //     .catch(error => {
+        //         Alert.alert("Error", "The video is corrupted. Please try again later", "OK");
```

Comment:
Comment if dead code like this, is supposed to be used in the future. Else delete

Reply:
Might be used in the future, thus we kept it commented out for now.